### DISCUSSION TITLE
Minimum build requirements for alpine-gcc?

### DISCUSSION THREAD

**Discussion by dhern023** [nlohmann/json#3155] [Q&A] [answered]
I'm building the develop branch with FetchContent using the CMakeLists.txt straight from the documentation. It works and builds properly with this docker image, my CMakeLists.txt, make etc.

```
FROM ubuntu:rolling AS linux

    RUN apt update && apt-get upgrade && apt-get install -y build-essential \
    git \
    cmake \
    autoconf \
    libtool \
    pkg-config
```

I'd like to build it from alpine-gcc to keep the image small.

```
FROM alpine:latest AS alpine_gcc

RUN apk update && \
    apk upgrade && \
    apk --update add \
        gcc \
        g++ \
        gdb \
        git \
        build-base \
        cmake \
        make \
        libtool \
        bash \
        libstdc++ \
        cppcheck \
        autoconf \
        libtool \
        pkg-config \
    rm -rf /var/cache/apk/*
```

However, I get something like this error
```
#10 12.51 /usr/include/c++/10.3.1/type_traits:906:12: error: invalid use of incomplete type 'struct std::__is_constructible_impl<nlohmann::basic_json<>, const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer>&>'
#10 12.51   906 |     struct is_constructible
```

Is there a list of minimum build requirements? I must be missing packages. but I'm not sure what.

**Comment by nlohmann** [comment, marked as answer]
You need a supported compiler (https://github.com/nlohmann/json#supported-compilers) and switch on C++11 mode.

**Reply by dhern023** [reply]
I see. I looked around and alpine is tricky for other packages too. I'll stick with my Ubuntu image.

**Reply by nlohmann** [reply]
I just checked your alpine Dockerfile. I could run the following commands without issue:

```sh
git clone https://github.com/nlohmann/json.git
cd json/
mkdir build
cd build/
cmake ..
make
ctest
```

Meaning the library is perfectly fine to be used in this setting. Please check your build settings, because it's not the Docker image that is causing the error.

**Reply by nlohmann** [reply]
How do you compile that file?

**Reply by dhern023** [reply]
Figured it out. With my CMakeLists.txt configuration, I needed 

```
#include <nlohmann/json.hpp> //not <json.hpp>

using json = nlohmann::json;
```

and it worked just fine.

Thanks
